### PR TITLE
[DUOS-1678][risk=no] consent update vote api fixes

### PR DESCRIPTION
**Pull request by shaemarks**
[Link to Jira ticket](https://broadworkbench.atlassian.net/browse/DUOS-1678)
Summary of changes:

- Modify updateVotes in VoteResource to take in vote, rationale, and vote ids in json format 
- Create custom model for json object 
- Convert tests to new method signature and add tests for when vote and rationale are null

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java [hunk 1]
```diff
@@ -87,10 +83,56 @@ public Response updateVotes(
...
+            List<Vote> updatedVotes = voteService.updateVotesWithValue(votes, voteUpdateInfo.getVote(), voteUpdateInfo.getRationale());
             return Response.ok().entity(updatedVotes).build();
         } catch (Exception e) {
             return createExceptionResponse(e);
         }
     }
+
+    static class VoteUpdateInfo {
+
+        @JsonProperty
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java]: As a side note, `@JsonProperty` is mostly unnecessary. This is a Jackson annotation that allows for writing/parsing via Jackson libraries. Since we're using Gson here, we don't need them. We should have a tech debt story to deprecate this usage.

File: src/main/resources/assets/paths/votes.yaml [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
     content:
       application/json:
         schema:
           type: string
-          example: [1, 2, 3]
+          example: {
+            "vote": true|false,
+            "rationale": "rationale string",
+            "voteIds": [1, 2, 3]
+          }
```
Comment (rushtong) [inline src/main/resources/assets/paths/votes.yaml:28]: Very nice!

File: src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java [hunk 3]
```diff
@@ -87,10 +83,56 @@ public Response updateVotes(
...
 
-            List<Vote> updatedVotes = voteService.updateVotesWithValue(votes, vote, rationale);
+            List<Vote> updatedVotes = voteService.updateVotesWithValue(votes, voteUpdateInfo.getVote(), voteUpdateInfo.getRationale());
             return Response.ok().entity(updatedVotes).build();
         } catch (Exception e) {
             return createExceptionResponse(e);
         }
     }
+
+    static class VoteUpdateInfo {
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java]: Thinking out loud here. We have this simple model class, and I have one for a similar method in my PR. It might make sense to pull these out into a separate model package specific to the resources directly. Another possibility would be in `org/broadinstitute/consent/http/models/vote`

**Review by rushtong — APPROVED** [review]
See minor comments inline, all are optional, address if you prefer. Looks good!

File: src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java [hunk 4]
```diff
@@ -1,12 +1,13 @@
 package org.broadinstitute.consent.http.resources;
 
 
+import com.fasterxml.jackson.annotation.JsonProperty;
```
Comment (JVThomas) [inline src/main/java/org/broadinstitute/consent/http/resources/VoteResource.java]: You can remove this import since you got rid of the `@JsonProperty` annotation

**Review by JVThomas — APPROVED** [review]
One little nitpick, but other than that the code looks good. Tests cover all of the expected edge cases and the functionality works when I tested it locally.
